### PR TITLE
Add CI & tests

### PR DESCRIPTION
**Pull request by inDream**
Connects likecoin/puttyimages-web#23

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -30,6 +30,12 @@ $ docker-compose up
 Local files are mount into docker you can modify file in your fs and will
 hot reload.
 
+## Test
+
+``` bash
+./run_test.sh
+```
+
```
Comment (rickmak) [inline README.md]: Please use Makefile as entry point instead of shell script.

File: __mocks__/ipfs-api.js [hunk 2]
```diff
@@ -0,0 +1,12 @@
+let mockFiles = {};
+
+const IPFSApi = {
+  setMockFiles(files) {
+    mockFiles = files;
+  },
+  files: {
+    cat: jest.fn().mockImplementation(path => Promise.resolve(mockFiles[path])),
```
Comment (rickmak) [inline __mocks__/ipfs-api.js]: Our prettierrc is `"arrowParens": "always",`

File: __tests__/integration/api/api.js [hunk 3]
```diff
@@ -0,0 +1,10 @@
+const axios = require('axios');
+// import axios from 'axios';
```
Comment (rickmak) [inline __tests__/integration/api/api.js]: Remove this comment.

File: __tests__/integration/api/api.js [hunk 4]
```diff
@@ -0,0 +1,10 @@
...
+
+const host = process.env.HOST || '127.0.0.1';
+const port = process.env.PORT || 3000;
+
+module.exports = {
+  get: url => axios.get(`http://${host}:${port}/api/${url}`),
```
Comment (rickmak) [inline __tests__/integration/api/api.js]: `"arrowParens": "always",`

File: server/models/index.js [hunk 5]
```diff
@@ -11,13 +11,14 @@ const sequelize = new Sequelize({
...
 const db = {};
+const isTest = global.test !== undefined;
 
 fs.readdirSync(__dirname).forEach((file) => {
   if (file === 'index.js' || file === 'base.js') {
     return;
   }
 
-  const model = sequelize.import(path.join('../', __dirname, file));
+  const model = sequelize.import(path.join(isTest ? '' : '../', __dirname, file));
```
Comment (rickmak) [inline server/models/index.js]: I prefer to make a variable `sequelizePath` right after L14. It help to group up the testing related logic in one place.

**Review by rickmak — COMMENTED** [review]
Please address the comments.

And the comments message of `ddbba21` should be `Expose the db port for testing in Travis`.

**Comment by inDream** [comment]
@rickmak updated, the travis fix is due to their image using port 5432 therefore cannot expose the same port

**Comment by rickmak** [comment]
I think this PR is mergeable.

The command less stateful will be better, to achieve this:

1. Remove the need of docker-compose up before integration test
  - Create a docker-compose.test.yml to speak out the testing dependency on integration test.
  - Use docker-compose run instead of exec to remove the assumption on the services is already up and running.
2. Separate the stateless unit test command away from integration test.

**Pull request merged**